src/document/mod.rs

```diff
@@ -84,113 +116,241 @@ impl DocumentEncryptOpts {
     fn default() -> Self {
         DocumentEncryptOpts::with_explicit_grants(None, None, true, vec![])
     }
 }
 
 #[async_trait]
 pub trait DocumentOps {
-    /// List all of the documents that the current user is able to decrypt.
+    /// Encrypts the provided document bytes.
     ///
-    /// # Returns
-    /// `DocumentListResult` struct with vec of metadata about each document the user can decrypt.
+    /// Returns a `DocumentEncryptResult` which contains document metadata as well as the `encrypted_data`,
+    /// which is the only thing that must be passed to [document_decrypt()](trait.DocumentOps.html#tymethod.document_decrypt)
+    /// in order to decrypt the document.
+    ///
+    /// Metadata about the document will be stored by IronCore, but the encrypted bytes of the document will not. To encrypt
+    /// without any document information being stored by IronCore, consider using
+    /// [document_encrypt_unmanaged()](advanced/trait.DocumentAdvancedOps.html#tymethod.document_encrypt_unmanaged) instead.
+    ///
+    /// # Arguments
+    /// - `document_data` - Bytes of the document to encrypt
+    /// - `encrypt_opts` - Document encryption parameters. Default values are provided with `DocumentEncryptOpts::default()`.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// use ironoxide::document::DocumentEncryptOpts;
+    /// let data = "secret data".as_bytes();
+    /// let encrypted = sdk.document_encrypt(data, &DocumentEncryptOpts::default()).await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_encrypt(
+        &self,
+        document_data: &[u8],
+        encrypt_opts: &DocumentEncryptOpts,
+    ) -> Result<DocumentEncryptResult>;
+
+    /// Decrypts an IronCore encrypted document.
+    ///
+    /// Returns details about the document as well as its decrypted bytes.
+    ///
+    /// # Arguments
+    /// - `encrypted_document` - Bytes of encrypted document. These should be the same bytes returned from
+    /// [document_encrypt()](trait.DocumentOps.html#tymethod.document_encrypt).
+    ///
+    /// # Errors
+    /// Fails if passed malformed data or if the calling user does not have sufficient access to the document.
+    ///
+    /// # Examples
+    /// ```
+    /// # async fn run() -> Result<(), ironoxide::IronOxideErr> {
+    /// # use ironoxide::prelude::*;
+    /// # let sdk: IronOxide = unimplemented!();
+    /// # let encrypted_data: Vec<u8> = vec![];
+    /// let decrypted_document = sdk.document_decrypt(&encrypted_data).await?;
+    /// let decrypted_data = decrypted_document.decrypted_data();
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn document_decrypt(&self, encrypted_document: &[u8]) -> Result<DocumentDecryptResult>;
+
+    /// Lists all of the encrypted documents that the calling user can read or decrypt.
```

Comment:
```suggestion
    /// Lists metadata for all of the encrypted documents that the calling user can read/decrypt.
```

Reply:
I added the "metadata for", but I prefer to keep the "or" over the slash